src/services/notification.service.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const { EventEmitter } = require('events');
+
+const eventBus = 'message';
+
+const WalletEventMap = {
```

Comment:
While I like the idea, this would remove the original event type from the map.
For instance, if we had an event `"foo-bar"` which we translate to `fooBar: 'wallet:foo'` a dev searching for instances of the event `"foo-bar"` may miss this reference.

What do you think of this?
Obs: my opinion is that i would still change to camel case, i just want to show the full info on why the original event name is there.

Reply:
I agree with you that the search by event name is the main way of finding the interactions with these events now, and that it's important to keep is searchable like this. As a short-term solution maybe just add the original event name as a comment?

```js
const ConnectionEventMap = {
  state: 'conn:state', // original lib event: 'state'
  newTx: 'wallet:new-tx', // original lib event: 'new-tx'
  updateTx: 'wallet:update-tx', // original lib event: 'update-tx'
```

Do you agree that this helps avoid confusion?

--
As a more definitive solution, I would expand my suggestion to also refactor the way events are referred on the project and on the wallet lib itself:

```js
// On wallet-lib/src/constants.js , add the following object:
  EVENT_NAMES: {
    STATE: 'state',
    NEW_TX: 'new-tx',
    UPDATE_TX: 'update-tx',
  }

// On headless/src/services/notification.service/js , simply re-use the same key names as the lib's constants object
const WalletEventMap = {
    STATE: 'wallet:state',
    NEW_TX: 'wallet:new-tx',
    UPDATE_TX: 'wallet:update-tx',
}

// The event listeners would then have the format:
import { EVENT_NAMES as LIB_EVENTS } from '@hathor/wallet-lib/constants'
hwallet.on(LIB_EVENTS.NEW_TX, tx => {
  const type = WalletEventMap.NEW_TX;
  ...
}
```

This seems material for another issue / PR on both projects, as I think it would make it simpler for developers to also use the lib's events beyond the external notifications.

Reply:
Created an [issue](https://github.com/HathorNetwork/hathor-wallet-lib/issues/425) to track this suggestion.